src/lambda/local/debugConfigurationProvider.ts

```diff
@@ -0,0 +1,265 @@
+                        request: 'attach',
+                        name: `Lambda: Attach to ${localLamdba.lambda.lambda} locally`,
+                        preLaunchTask: undefined,
+                        address: 'localhost',
+                        port: 5858,
+                        localRoot,
+                        remoteRoot: '/var/task',
+                        protocol: 'inspector',
+                        skipFiles: [
+                            '/var/runtime/node_modules/**/*.js',
+                            '<node_internals>/**/*.js'
+                        ]
+                    }
+                )
+
+                return accumulator
+            },
+            []
+        )
+    }
+
+    /**
+     * `sam init` puts the local root in a subdirectory. We attempt to detect this subdirectory by looking
+     * for child folders that contain a package.json file. If the root workspace folder does not contain
+     * package.json, AND exactly one of its direct children contains package.json, use that child as the
+     * local root.
+     */
+    private async findNpmProject(
+        folder: WorkspaceFolder,
+        token?: CancellationToken
+    ): Promise<string | undefined> {
+        // The root directory is an npm package, so we don't need to look in subdirectories.
+        if (await fileExists(join(folder.uri.fsPath, 'package.json'))) {
+            return undefined
+        }
+
+        const entries: string[] = await readdirAsync(folder.uri.fsPath)
+
+        const candidates: string[] = (await Promise.all(entries.map(async entry => {
+            const entryPath = join(folder.uri.fsPath, entry)
+            if (await fileExists(entryPath) && (await statAsync(entryPath)).isDirectory()) {
+                return await fileExists(join(entryPath, 'package.json')) ? entry : undefined
+            }
+
+            return undefined
+        }))).filter(c => !!c).map(c => c as string)
+
+        return candidates.length === 1 ? candidates[0] : undefined
+    }
+
+    private getTaskLabel(functionName: string): string {
+        return `Lambda: Invoke ${functionName} locally`
+    }
+
+    private async addPreLaunchTask(
+        folder: WorkspaceFolder,
+        functionName: string,
+        event: { [ key: string ]: string } = {},
+        debugPort: number = 5858
+    ): Promise<string> {
+        const label = this.getTaskLabel(functionName)
+        const configRoot = join(folder.uri.fsPath, '.vscode')
+        const tasksPath = join(configRoot, 'tasks.json')
+
+        let tasks: TasksConfig | undefined
+        if (await fileExists(tasksPath)) {
+            tasks = JSON.parse(await readFileAsString(tasksPath, 'utf8')) as TasksConfig | undefined
+        }
+
+        if (!tasks) {
+            tasks = {
+                version: '2.0.0'
+            }
+        }
+
+        if (!tasks.tasks) {
+            tasks.tasks = []
+        }
+
+        if (!tasks.tasks.some(t => t.label === label)) {
```

Comment:
I don't think so. If the task differs, it's probably because the user customized it, and we don't want to overwrite their customizations. The other possibility is that the user coincidentally added a task with the label `Lambda: Debug MyFunction Locally`, but I don't think that this will be a common case, and we can't distinguish it from 'user customized the task that we generated'.